src/v/datalake/values_parquet.cc

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "datalake/values_parquet.h"
+
+namespace datalake {
+namespace {
+struct primitive_value_converting_visitor {
+    parquet_conversion_outcome operator()(iceberg::boolean_value v) {
+        return serde::parquet::boolean_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::int_value v) {
+        return serde::parquet::int32_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::long_value v) {
+        return serde::parquet::int64_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::float_value v) {
+        return serde::parquet::float32_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::double_value v) {
+        return serde::parquet::float64_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::date_value v) {
+        return serde::parquet::int32_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::time_value v) {
+        return serde::parquet::int64_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::timestamp_value v) {
+        return serde::parquet::int64_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::timestamptz_value v) {
+        return serde::parquet::int64_value{.val = v.val};
+    }
+    parquet_conversion_outcome operator()(iceberg::string_value v) {
+        return serde::parquet::byte_array_value{.val = std::move(v.val)};
+    }
+    parquet_conversion_outcome operator()(iceberg::uuid_value v) {
+        iobuf buffer;
+        buffer.append(v.val.uuid().data, v.val.uuid().size());
+        return serde::parquet::fixed_byte_array_value{.val = std::move(buffer)};
+    }
+    parquet_conversion_outcome operator()(iceberg::fixed_value v) {
+        return serde::parquet::fixed_byte_array_value{.val = std::move(v.val)};
+    }
+    parquet_conversion_outcome operator()(iceberg::binary_value v) {
+        return serde::parquet::byte_array_value{.val = std::move(v.val)};
+    }
+    parquet_conversion_outcome operator()(iceberg::decimal_value v) {
+        iobuf buffer;
+        // big endian, starts from msb
+        auto low_part = ss::cpu_to_be(Int128Low64(v.val));
+        auto high_part = ss::cpu_to_be(Int128High64(v.val));
+        buffer.append(
+          reinterpret_cast<const char*>(&high_part), sizeof(int64_t));
+        buffer.append(
+          reinterpret_cast<const char*>(&low_part), sizeof(uint64_t));
```

Comment:
why is this sizeof different than the other?

Reply:
the value is the same, it is done for consistency, the low part of `absl::in128` is unsigned, while the high part is signed 64-bit integer

Reply:
gotcha. thanks